src/ConnectedMode/UI/OrganizationSelection/OrganizationSelectionDialog.xaml.cs

```diff
@@ -20,42 +20,75 @@
     {
+        this.connectedModeServices = connectedModeServices;
         ViewModel = new OrganizationSelectionViewModel(credentialsModel, connectedModeServices.SlCoreConnectionAdapter, new ProgressReporterViewModel());
         InitializeComponent();
     }
 
     public OrganizationSelectionViewModel ViewModel { get; }
 
-    private void OkButton_OnClick(object sender, RoutedEventArgs e)
+    private async void OkButton_OnClick(object sender, RoutedEventArgs e)
     {
-        DialogResult = true;
+        var isConnectionValid = await ValidateConnectionForSelectedOrganizationAsync(ViewModel.SelectedOrganization.Key);
+        if(isConnectionValid)
+        {
+            OnConnectionValidationSucceeded(ViewModel.SelectedOrganization.Key);
+        }
     }
 
-    private void ChooseAnotherOrganizationButton_OnClick(object sender, RoutedEventArgs e)
+    private async void ChooseAnotherOrganizationButton_OnClick(object sender, RoutedEventArgs e)
     {
         ViewModel.SelectedOrganization = null;
         var manualOrganizationSelectionDialog = new ManualOrganizationSelectionDialog();
-        manualOrganizationSelectionDialog.Owner = this;
-        var manualSelection = manualOrganizationSelectionDialog.ShowDialog(this);
-        ViewModel.SelectedOrganization =new OrganizationDisplay(manualOrganizationSelectionDialog.ViewModel.OrganizationKey, null);
-        if (manualSelection is true)
+        var manualSelectionDialogSucceeded = manualOrganizationSelectionDialog.ShowDialog(this);
+        if(manualSelectionDialogSucceeded is not true)
+        {
+            return;
+        }
+
+        var isSelectedManualOrganizationValid = await ValidateConnectionForSelectedOrganizationAsync(manualOrganizationSelectionDialog.ViewModel.OrganizationKey);
+        if (isSelectedManualOrganizationValid)
+        {
+            OnConnectionValidationSucceeded(manualOrganizationSelectionDialog.ViewModel.OrganizationKey);
+        }
+    }
+
+    private async Task<bool> ValidateConnectionForSelectedOrganizationAsync(string selectedOrganizationKey)
+    {
+        try
+        {
+            var organizationSelectionInvalidMsg = string.Format(UiResources.ValidatingOrganziationSelectionFailedText, selectedOrganizationKey);
+            return await ViewModel.ValidateConnectionForOrganizationAsync(selectedOrganizationKey, organizationSelectionInvalidMsg);
+        }
+        catch (Exception e) when (!ErrorHandler.IsCriticalException(e))
         {
-            DialogResult = true;
+            connectedModeServices.Logger.WriteLine(e.ToString());
+            return false;
         }
     }
 
     private async void OrganizationSelectionDialog_OnLoaded(object sender, RoutedEventArgs e)
     {
         await ViewModel.LoadOrganizationsAsync();
     }
+
+    private void OnConnectionValidationSucceeded(string organizationKey)
+    {
+        ViewModel.UpdateConnectionInfo(organizationKey);
```

Comment:
why not call this inside the validation method and use the ConnectionInfo object that was already created there?

Reply:
Because it should only be done once the validation succeeded and the dialog is closed.